tests/htmllib.py

```diff
@@ -49,42 +49,16 @@ def start_element (self, tag, attrs, element_text, lineno, column):
         @type attrs: dict
         @return: None
         """
-        self._start_element(tag, attrs, ">", element_text)
-        self.fd.write("</%s>" % tag)
-
-    def start_end_element (self, tag, attrs, element_text, lineno, column):
-        """
-        Print HTML start-end element.
-
-        @param tag: tag name
-        @type tag: string
-        @param attrs: tag attributes
-        @type attrs: dict
-        @return: None
-        """
-        self._start_element(tag, attrs, "/>", element_text)
-
-    def _start_element (self, tag, attrs, end, element_text):
-        """
-        Print HTML element with end string.
-
-        @param tag: tag name
-        @type tag: string
-        @param attrs: tag attributes
-        @type attrs: dict
-        @param end: either > or />
-        @type end: string
-        @return: None
-        """
         self.fd.write("<%s" % tag.replace("/", ""))
         for key, val in sorted(attrs.items()):
             if val is None:
                 self.fd.write(" %s" % key)
             else:
                 self.fd.write(' %s="%s"' % (key, quote_attrval(val)))
-        self.fd.write(end)
         if element_text:
-            self.fd.write(element_text)
+            self.fd.write(">%s</%s>" % (element_text, tag))
+        else:
+            self.fd.write("/>")
```

Comment:
this seems strange - why do we need to change this behaviour here? not that i object, it makes sense to do <foo/> instead of <foo></foo>, but it seems gratuitous.

Reply:
Makes sense to me: the internal handler protocol changed -- the parser no longer calls end_element or start_end_element, there's only start_element, so now it needs to do the job of all three.

Reply:
See review comment below:
https://github.com/linkchecker/linkchecker/pull/360#discussion_r406675103